chain-factory/src/factory.rs

```diff
@@ -42,9 +61,38 @@ pub trait FactoryModule {
         let _ = self.all_deployed_contracts().swap_remove(&sc_address);
     }
 
+    #[endpoint(addContractsToMap)]
+    fn add_contracts_to_map(
+        &self,
+        contracts_list: MultiValueEncoded<Self::Api, ContractMap<Self::Api>>,
+    ) {
+        for contract in contracts_list {
+            require!(
+                self.blockchain().is_smart_contract(&contract.address),
+                "Given contract address is not a correct SC address"
+            );
+
+            require!(
+                self.is_required_contract_name(&contract.name),
+                "Given contract name is not a required SC"
+            );
+
+            self.contracts_map(contract.name).set(contract.address);
+        }
+    }
+
+    fn is_required_contract_name(&self, contract_name: &ManagedBuffer) -> bool {
```

Comment:
`#[inline]` is a rust annotation that it's used for efficiency, basically instead of calling the function it actually replaces the call with the body of the function
https://matklad.github.io/2021/07/09/inline-in-rust.html
At the moment I don't think it's necessary to inline it since it maps over the `SC_ARRAY`